flink-batch-connectors/flink-avro/src/main/java/org/apache/flink/api/java/io/AvroInputFormat.java

```diff
@@ -119,12 +138,14 @@ public E nextRecord(E reuseValue) throws IOException {
 		if (reachedEnd()) {
 			return null;
 		}
-		
-		if (!reuseAvroValue) {
-			reuseValue = InstantiationUtil.instantiate(avroValueType, Object.class);
+		if (org.apache.avro.generic.GenericRecord.class == avroValueType) {
+			return reuseAvroValue ? dataFileReader.next(reuseValue) : dataFileReader.next();
+		} else {
+			if (!reuseAvroValue) {
+				reuseValue = InstantiationUtil.instantiate(avroValueType, Object.class);
+			}
+			reuseValue = dataFileReader.next(reuseValue);
```

Comment:
The use of the ternary operator seems to be very inconsistent here.
Either use it for the return value and the assignment or not at all.

I personally prefer to avoid it whenever possible.